alternator/CMakeLists.txt

```diff
@@ -17,6 +17,7 @@ target_sources(alternator
     streams.cc
     consumed_capacity.cc
     ttl.cc
+    expression_factory.cc
```

Comment:
A thought while reading this:
The "expression factory" you created is not really an expression factor - it doesn't create expressions. Rather, it creates "parsed expressions" (one of the alternator::parsed::* types) so parsed_expression_factory, or more succinctly - it "parses" stuff. So perhaps it should be an expression_parser?